### PR TITLE
{AKS} Refactor 11th part of the update sub-command in acs module

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

- Refactor updating identity profile.

Unit Test Coverage:

> "covered_lines": 1738,
> "num_statements": 1754,
> "percent_covered": 99.08779931584948,
> "missing_lines": 16,
> "excluded_lines": 0

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -5563,6 +5561,60 @@ def update_auto_upgrade_profile(self, mc: ManagedCluster) -> ManagedCluster:
             mc.auto_upgrade_profile.upgrade_channel = auto_upgrade_channel
         return mc
 
+    def update_identity(self, mc: ManagedCluster) -> ManagedCluster:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:5564]: The implementation of this function basically follows [this part](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/custom.py#L2656-L2694) in file `custom.py`.

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 2]
```diff
@@ -1754,23 +1754,25 @@ def _get_enable_managed_identity(
         """
         # read the original value passed by the command
         enable_managed_identity = self.raw_param.get("enable_managed_identity")
-        # try to read the property value corresponding to the parameter from the `mc` object
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:1757]: Currently we are refactoring `aks create` and `aks update`, and "In create mode" means this only applies to the case of `aks create` but not `aks update`.

Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:1757]: Currently we are refactoring `aks create` and `aks update`, and "In create mode" means this only applies to the case of `aks create` but not `aks update`.

**Review by norshtein — APPROVED** [review]

**Comment by yonzhan** [comment]
AKS

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**